public/static/app/perfdiscovery/PerfDiscoveryStateService.js

```diff
@@ -0,0 +1,168 @@
+mciModule.factory('PerfDiscoveryStateService', function(
+  $httpParamSerializer, $location
+) {
+  /* STATE ADAPTORS */
+
+  function sortToState(cols) {
+    return {
+      sort: _.reduce(cols, function(m, d) {
+        m[d.field] = d.sort
+        return m
+      }, {})
+    }
+  }
+
+  function onSortChanged(state) {
+    return function(grid, cols) {
+      applyState(state, sortToState(cols))
+    }
+  }
+
+  function filteringToState(cols) {
+    return {
+      filtering: _.reduce(cols, function(m, d) {
+        m[d.field] = d.filters[0].term
+        return m
+      }, {})
+    }
+  }
+
+  function onFilteringChanged(state, grid) {
+    return function() {
+      applyState(
+        state,
+        filteringToState(
+          _.filter(
+            grid.columns, function(col) { return col.filters[0].term }
+          )
+        )
+      )
+    }
+  }
+
+  /* SERIALIZATION */
+
+  function serializeFromVersion(state) {
+    return {from: state.from}
+  }
+
+  function serializeToVersion(state) {
+    return {to: state.to}
+  }
+
+  function serealizeSorting(state) {
+    return {
+      sort: _.map(state.sort, function(v, k) {
+        return (v.direction == 'desc' ? '-' : '') + k + v.priority
```

Comment:
I'd prefer to use ===

Reply:
In general, yes, `===` is preferable. However for comparing strings it doesn't matter. I found `===` syntax a bit annoying and prefer when it hidden (like in CoffeeScript).